### ISSUE TITLE
0003376:
use favorites concept in felamimail

### ISSUE THREAD

**Issue by Gloirin** [tine20/Tine-2.0-Open-Source-Groupware-and-CRM#1765]
**Reported by pschuele on 22 Nov 2010 13:58**

use favorites concept in felamimail

- add favorites to tree panel
- make message search account-independent
- add more filters
- make saving of favorites work

**Comment by Gloirin** [comment]
Depends on #1743

**Comment by Gloirin** [comment]
Depends on #877

**Comment by Gloirin** [comment]
Depends on #1781

**Comment by Gloirin** [comment]
**Comment posted by robert.lischke on 23 Nov 2010 12:24**

Ok, so the &quot;favorites&quot; appear above all Inboxes, just like in every other application

Filtering should be possible by
a) Date/Sent/Received -- before, equal, is, after, etc.
b) Subject -- contains
c) Flags -- yes/no
d) From (Email) -- contains, is, etc.
e) From (Name) -- contains, is, etc.
f) Attachments -- has attachment/no attachment
g) To -- contains, is, etc.
and possibly
h) Sender -- This appears in the svn-trunk, What exactly is this? This column seems to be empty.

I think &quot;Id&quot; and &quot;Size&quot; are not relevant.

Default Favorites could be (in this order!)
- All Inboxes
- All highlighted mail
- All unread mail (Exclude Spam-folder? Is this possible?)
- All Drafts

Drag&amp;Drop of Mail from the favorites to any folder _must_ be possible, this would include uploading mail from one account to another!

**Comment by Gloirin** [comment]
**Comment posted by robert.lischke on 23 Nov 2010 12:43**

Automatic reloading the favorite should be consistent to the rest of Tine, i.e.
- after drag&amp;drop
- after opening/closing a mail
- explicit reload through button

This gets tricky when using the &quot;All unread mail&quot; default favorite and leads to mail disappearing (since it has been &quot;read&quot;) although users might want to perform actions with the mail -- therefore I think we should change the behavior of when a mail is marked as read. Will open a new ticket.

**Comment by Gloirin** [comment]
**Comment posted by pschuele on 24 Nov 2010 11:42**

note 3302 -&gt; agree
note 3304 -&gt; disagree: we do not want to reload the mail grid often. perhaps we&#039;ll roll out this behavior to the other apps

**Comment by Gloirin** [comment]
**Comment posted by robert.lischke on 24 Nov 2010 12:02**

Ok, Note 3304 was just based on an observation and the important part is the disappearance of mails after being automatically marked as read :)

Will check as soon as it is in the trunk!
